Windows failing test: k8s.io/kubernetes/pkg/kubelet/volumemanager/reconciler TestReconstructedVolumeShouldUnmountSucceedAfterSetupFailed

### Failure cluster [296948d05b810f4e2286](https://go.k8s.io/triage#296948d05b810f4e2286)

##### Error text:
```
Failed
=== RUN   TestReconstructedVolumeShouldUnmountSucceedAfterSetupFailed
    testing.go:1369: TempDir RemoveAll cleanup: unlinkat C:\Users\azureuser\AppData\Local\Temp\TestReconstructedVolumeShouldUnmountSucceedAfterSetupFailed490003075\001: The directory is not empty.
--- FAIL: TestReconstructedVolumeShouldUnmountSucceedAfterSetupFailed (0.01s)

```
#### Recent failures:
[3/5/2026, 10:09:16 AM pr:pull-kubernetes-unit-windows-master](https://prow.k8s.io/view/gs/kubernetes-ci-logs/pr-logs/pull/137423/pull-kubernetes-unit-windows-master/2029619892440797184)
[3/5/2026, 6:52:02 AM pr:pull-kubernetes-unit-windows-master](https://prow.k8s.io/view/gs/kubernetes-ci-logs/pr-logs/pull/137361/pull-kubernetes-unit-windows-master/2029570242551746560)
[3/4/2026, 5:14:44 PM pr:pull-kubernetes-unit-windows-master](https://prow.k8s.io/view/gs/kubernetes-ci-logs/pr-logs/pull/136356/pull-kubernetes-unit-windows-master/2029364522296283136)
[3/4/2026, 3:13:13 PM pr:pull-kubernetes-unit-windows-master](https://prow.k8s.io/view/gs/kubernetes-ci-logs/pr-logs/pull/137420/pull-kubernetes-unit-windows-master/2029334002237706240)
[3/4/2026, 1:31:38 PM pr:pull-kubernetes-unit-windows-master](https://prow.k8s.io/view/gs/kubernetes-ci-logs/pr-logs/pull/137357/pull-kubernetes-unit-windows-master/2029308413808218112)


/kind failing-test
<!-- If this is a flake, please add: /kind flake -->

<!-- Please assign a SIG using: /sig SIG-NAME -->
/kind flake
/sig windows

--- a/pkg/kubelet/volumemanager/reconciler/reconciler_test.go
+++ b/pkg/kubelet/volumemanager/reconciler/reconciler_test.go
@@ -2017,10 +2017,14 @@ func waitForUnmount(
 	err := retryWithExponentialBackOff(
 		testOperationBackOffDuration,
 		func() (bool, error) {
-			if asw.PodHasMountedVolumes(podName) {
-				return false, nil
+			// GetAllMountedVolumes includes both VolumeMounted and VolumeMountUncertain
+			// states, so this correctly detects unmount for reconstructed (uncertain)
+			// volumes as well as normally mounted volumes.
+			for _, v := range asw.GetAllMountedVolumes() {
+				if v.PodName == podName {
+					return false, nil
+				}
 			}
-
 			return true, nil
 		},
 	)
@@ -2586,6 +2590,21 @@ func TestReconstructedVolumeShouldUnmountSucceedAfterSetupFailed(t *testing.T) {
 
 	// Act first reconcile to trigger mount reconstructed volume
 	reconciler.reconcile(ctx)
+
+	// Wait for the async mount operation to actually complete before proceeding.
+	// The mount runs in a goroutine and creates directories on disk. We must
+	// wait for it to finish so the subsequent unmount is not blocked by a
+	// pending mount operation.
+	err = retryWithExponentialBackOff(
+		testOperationBackOffDuration,
+		func() (bool, error) {
+			return volumetesting.VerifySetUpCallCount(1, fakePlugin) == nil, nil
+		},
+	)
+	if err != nil {
+		t.Fatalf("Timed out waiting for mount operation to complete")
+	}
+
 	waitForUncertainPodMount(t, generatedVolumeName, podName, asw)
 
 	// mock remove pod

